include/deal.II/lac/linear_operator.h

```diff
@@ -0,0 +1,632 @@
+ *   dealii::Vector<Number>::size_type n() const;
+ *
+ *   // Application of matrix to vector src. Writes result into dst
+ *   vmult(dealii::Vector<Number> &dst, const dealii::Vector<Number> &src);
+ *
+ *   // Application of transpose of matrix to vector src. Writes result
+ *   // into dst. (Depending on the usage of the linear operator class this
+ *   // can be a dummy implementation throwing an error)
+ *   Tvmult(dealii::Vector<Number> &dst, const dealii::Vector<Number> &src);
+ * };
+ * @endcode
+ *
+ * @authors: Matthias Maier, 2015
+ */
+template <template <typename> class Matrix, typename Number>
+LinearOperator<dealii::Vector<Number>, dealii::Vector<Number> >
+linop(const Matrix<Number> &matrix)
+{
+  // Store all information about matrix in the lambda functions by
+  // reference to ensure that a modification of matrix after the creation
+  // of a LinearOperator wrapper is respected.
+
+  LinearOperator<dealii::Vector<Number>, dealii::Vector<Number>> return_op;
+
+  return_op.new_range_member = [&matrix]() -> dealii::Vector<Number>
+  {
+    return dealii::Vector<Number> (matrix.m());
+  };
+
+  return_op.new_domain_member = [&matrix]() -> dealii::Vector<Number>
+  {
+    return dealii::Vector<Number> (matrix.n());
+  };
+
+  return_op.vmult =
+    [&matrix](dealii::Vector<Number> &v, const dealii::Vector<Number> &u)
+  {
+    Assert(v.size() == matrix.m(),
+           dealii::ExcDimensionMismatch(v.size(), matrix.m()));
+    Assert(u.size() == matrix.n(),
+           dealii::ExcDimensionMismatch(u.size(), matrix.n()));
+
+    matrix.vmult(v,u);
+  };
+
+  return_op.vmult_add =
+    [&matrix](dealii::Vector<Number> &v, const dealii::Vector<Number> &u)
+  {
+    Assert(v.size() == matrix.m(),
+           dealii::ExcDimensionMismatch(v.size(), matrix.m()));
+    Assert(u.size() == matrix.n(),
+           dealii::ExcDimensionMismatch(u.size(), matrix.n()));
+
+    matrix.vmult_add(v,u);
+  };
+
+  return_op.Tvmult =
+    [&matrix](dealii::Vector<Number> &v, const dealii::Vector<Number> &u)
+  {
+    Assert(v.size() == matrix.n(),
+           dealii::ExcDimensionMismatch(v.size(), matrix.n()));
+    Assert(u.size() == matrix.m(),
+           dealii::ExcDimensionMismatch(u.size(), matrix.m()));
+
+    matrix.Tvmult(v,u);
+  };
+
+  return_op.Tvmult_add =
+    [&matrix](dealii::Vector<Number> &v, const dealii::Vector<Number> &u)
+  {
+    Assert(v.size() == matrix.n(),
+           dealii::ExcDimensionMismatch(v.size(), matrix.n()));
+    Assert(u.size() == matrix.m(),
+           dealii::ExcDimensionMismatch(u.size(), matrix.m()));
+
+    matrix.Tvmult_add(v,u);
+  };
+
+  return return_op;
+}
+
+
+/**
+ * Returns a LinearOperator that is the identity of the vector space
+ * formed by all dealii::Vector<Number> objects of size @p size.
+ */
+template <typename Number>
+LinearOperator<dealii::Vector<Number>, dealii::Vector<Number> >
+linop_identity(typename dealii::Vector<Number>::size_type size)
```

Comment:
Which also doesn't help much. A generic identity function would need to get a LinearOperator exemplar to extract the <code>new_domain/range_member</code> function.

I refrained from doing this because the syntax is awkward:
```
linop_identity(op_a) - op_a
```